src/store/names.ts

```diff
@@ -0,0 +1,87 @@
+import { selector } from 'recoil';
+import { getStxAddress } from '@stacks/wallet-sdk';
+
+import { fetcher } from '@common/wrapped-fetch';
+
+import { accountsStore } from '@store/wallet';
+import { currentNetworkStore, currentTransactionVersion } from '@store/networks';
+
+async function fetchNamesByAddress(networkUrl: string, address: string): Promise<string[]> {
+  const res = await fetcher(networkUrl + `/v1/addresses/stacks/${address}`);
+  const data = await res.json();
+  return data?.names || [];
+}
+
+function makeKey(networkUrl: string, address: string): string {
+  return `${networkUrl}__${address}`;
+}
+
+function getLocalNames(networkUrl: string, address: string): string[] | null {
+  const key = makeKey(networkUrl, address);
+  const value = localStorage.getItem(key);
+  return value ? JSON.parse(value) : null;
+}
+
+function setLocalNames(networkUrl: string, address: string, names: string[]): void {
+  const key = makeKey(networkUrl, address);
+  return localStorage.setItem(key, JSON.stringify(names));
+}
+
+type AccountNameState =
+  | {
+      address: string;
+      index: number;
+      names: string[];
+    }[]
+  | null;
+
+export const accountNameState = selector<AccountNameState>({
+  key: 'names',
+  get: async ({ get }) => {
+    const accounts = get(accountsStore);
+    const network = get(currentNetworkStore);
+    const transactionVersion = get(currentTransactionVersion);
+
+    if (!network || !accounts) return null;
+
+    const promises = accounts?.map(async account => {
```

Comment:
```suggestion
    const promises = accounts.map(async account => {
```
`?` unnecessary owing to line 45?